Fix: Cast numeric write-concern values from XML mapping to integers

ikallali opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | bug
| BC Break     | no
| Fixed issues | Fix #2749

#### Summary

<!-- Provide a summary your change. -->

When using write-concern="1" in XML mapping, the value was being cast to string, causing MongoDB to interpret it as a named write concern mode instead of a numeric level. This led to errors like: "No write concern mode named '1' found in replica set configuration".

This commit ensures that numeric write concern values are properly cast to integers, while still allowing named modes such as "majority" to be used as strings.

ikallali committed:
Fix: Cast numeric write-concern values from XML mapping to integers

When using write-concern="1" in XML mapping, the value was being cast to string,
causing MongoDB to interpret it as a named write concern mode instead of a numeric level.
This led to errors like: "No write concern mode named '1' found in replica set configuration".

This commit ensures that numeric write concern values are properly cast to integers,
while still allowing named modes such as "majority" to be used as strings.

ikallali committed:
Merge pull request #1 from ikallali/fix/xml-write-concern-cast

Fix: Cast numeric write-concern values from XML mapping to integers

GromNaN commented:
In fact, the `int` value is accepted in the [attribute configuration](https://github.com/doctrine/mongodb-odm/blob/11198c2d29be2ec9ae45313fe32e7665f50bbf6e/lib/Doctrine/ODM/MongoDB/Mapping/Annotations/Document.php#L37). Since using  an `int` for this setting is [documented in MongoDB](https://www.mongodb.com/docs/manual/reference/write-concern/#w-option), let's make it possible to use it in XML.

Could you add a test to cover this feature?

ikallali committed:
Fix XML writeConcern parsing with numeric support

fixes code style issues to comply with doctrine/coding-standard:
- Adds space after `if` keyword
- Adds blank line after `if` block

ikallali committed:
Update XmlDriver.php

Fix CS issues in XmlDriver (writeConcern block)

- Remove empty space after `if` block

GromNaN committed:
Fix tests to be strict on WriteConcern type

GromNaN reviewed (approved):
Thank you @ikallali.
I fixed the tests to strictly check the write concern type; that confirmed the need for your patch.
